src/wallet/wallet.cpp

```diff
@@ -2720,14 +2720,36 @@ bool CWallet::CreateCoinStake(
     return true;
 }
 
+std::string CWallet::CommitResult::ToString() const
+{
+    std::string strErrRet = strprintf(_("Failed to accept tx in the memory pool (reason: %s)\n"), FormatStateMessage(state));
+
+    switch (status) {
+        case CWallet::CommitStatus::OK:
+            return _("No error");
+        case CWallet::CommitStatus::Abandoned:
+            strErrRet += _("Transaction canceled.");
+            break;
+        case CWallet::CommitStatus::NotAccepted:
+            strErrRet += strprintf(_("WARNING: The transaction has been signed and recorded, so the wallet will try to re-send it. "
+                    "Use 'abandontransaction' to cancel it. (txid: %s)"), hashTx.ToString());
+            break;
+        default:
+            return _("Invalid status error.");
+    }
+
+    return strErrRet;
+}
+
 /**
  * Call after CreateTransaction unless you want to abort
  */
-bool CWallet::CommitTransaction(CWalletTx& wtxNew, CReserveKey& reservekey, std::string strCommand)
+void CWallet::CommitTransaction(CWalletTx& wtxNew, CReserveKey& reservekey, CWallet::CommitResult& res, std::string strCommand)
 {
+    res.status = CWallet::CommitStatus::NotAccepted;
```

Comment:
This is not entirely true. Edge scenario, `AddToWallet` can return false if the disk write fail.

Reply:
Quite remote possibility. Anyway, the return value of  `AddToWallet` was never considered inside `CommitTransaction` (and even if writing to disk fails in `AddToWallet`, the mapWallet and mapTxSpends are still updated). We can address this in a separate PR (if we really want) as this one is intended to fix a specific issue (ATMP failure) unrelated to that.

Reply:
yep, all good. Future work mention.